src/Position.sol

```diff
@@ -0,0 +1,140 @@
+ */
+contract Position is Ownable {
+    using BasketLib for Basket;
+    using PriceConverter for Basket;
+    // uint256 public immutable minRatio;
+    // PriceFeed private immutable priceFeed;
+    Coin private immutable coin;
+    uint256 public debt;
+    bool isInsolvent;
+    Basket collateral;
+    uint256 public constant RATIO = 150;
+    uint256 public constant COLLATERAL_DECIMAL = 1e18;
+    address user;
+    uint256 public constant FEE = 500;
+    uint256 public constant PENALTY = 500;
+    uint256 private s_lastTimeStamp;
+
+    constructor(
+        IERC20[] memory tokens,
+        uint256[] memory weights,
+        AggregatorV3Interface[] memory priceFeeds,
+        address _coinAddress,
+        address _owner
+    ) {
+        uint256 length = tokens.length;
+        console.log(length);
+        for (uint256 i = 0; i < length; ++i) {
+            collateral.add(tokens[i], weights[i], priceFeeds[i]);
+        }
+        coin = Coin(_coinAddress);
+        transferOwnership(_owner);
+        user = _owner;
+        debt = 0;
+        s_lastTimeStamp = block.timestamp;
+        isInsolvent = false;
+    }
+
+    /**
+     *
+     * @dev calculate the total collateral amount in USD needed to hedge against
+     *      the total exposure
+     */
+    function calculateCollateralTotalAmount(
+        uint256 _stablecoinAmount
+    ) public pure returns (uint256) {
+        return (_stablecoinAmount * RATIO) / 100;
+    }
+
+    // function calculateCollateralTotalAmount(
+    //     uint256 _stablecoinAmount
+    // ) public view returns (uint256) {
+    //     return
+    //         (collateral.getConversionRateOfBasket(_stablecoinAmount) * RATIO) /
+    //         100;
+    // }
+
+    function TotalBalanceInUsd() public view returns (uint256) {
+        return collateral.Balance(address(this));
+    }
+
+    /**
+     * @dev Returns true if the contract's debt position can increase by a specified amount.
+     */
+    function canTake(uint256 _moreDebt) public view returns (bool) {
+        require(_moreDebt > 0, "Cannot take 0");
+        uint256 totalCollateralInUSD = TotalBalanceInUsd();
+        console.log(totalCollateralInUSD);
+        uint256 debtInCoins = debt + _moreDebt;
+        return ((totalCollateralInUSD * 100) / debtInCoins) >= RATIO;
+    }
+
+    /**
+     * @dev Takes out loan against collateral if the vault is solvent
+     *  An approve function needs to be added in case of liquidation
+     */
+    function take(address _receiver, uint256 _moreDebt) public onlyOwner {
+        require(canTake(_moreDebt), "Position cannot take debt");
+        coin.mint(address(this), _receiver, _moreDebt);
+        debt = debt + _moreDebt;
+    }
+
+    /**
+     * @dev Liquidates vault if priceId points to a price record of insolvency.
+     * This function should be in the notary contract with regular checks
+     */
+    function liquidate(uint256 priceId) public {
```

Comment:
This could use a large comment above explaining the algorithm of the liquidation.